src/Microsoft.DotNet.Compiler.Common/Executable.cs

```diff
@@ -97,13 +97,12 @@ private void MakeCompilationOutputRunnableForCoreCLR()
         {   
             WriteDepsFileAndCopyProjectDependencies(_exporter);
 
-            var emitEntryPoint = _compilerOptions.EmitEntryPoint ?? false;
+            var isRunnable = _compilerOptions.EmitEntryPoint ?? _context.ProjectFile.OverrideIsRunnable;
 
-            if (emitEntryPoint && !_context.IsPortable)
+            if (isRunnable && !_context.IsPortable)
             {
-                // TODO: Pick a host based on the RID
                 VerifyCoreClrPresenceInPackageGraph();
-                CoreHost.CopyTo(_runtimeOutputPath, _compilerOptions.OutputName + Constants.ExeSuffix);
```

Comment:
For publish, it would auto output hte host when it was in the package graph, the same is not true for build (or at least so I thought). I had expected there would be changes to tell build, go copy this one thing from the package graph.

I might be incorrect here, just want to have the right understanding.

---
In reply to: [71798032](https://github.com/dotnet/cli/pull/3911#discussion_r71798032) [](ancestors = 71798032,71797819,71797731,71797574)